library/init/meta/interactive.lean

```diff
@@ -521,6 +517,25 @@ do x ← mk_fresh_name,
    t ← get_local x,
    induction (to_pexpr t) rec_name hs ([] : list name)
 
+private meta def cases_arg_p : parser (option name × pexpr) :=
+with_desc "(id :)? expr" $ do
+  t ← texpr,
+  match t with
+  | (local_const x _ _ _) :=
```

Comment:
I'm not saying this is the best syntax I can think of, but I do believe that we should be as consistent as possible. Note that in term-mode `let` the name is not optional, which feels relatively intuitive to me.

In short, in term mode we have the syntaxes
```
"assume" binders
"assume" ":" expr -- anoymous
"have" ident ":" expr
"have" expr -- anonymous
"let" ident (":" expr)
```
With your trick, we emulate all of them in tactic mode.